homeassistant/helpers/device_registry.py

```diff
@@ -975,6 +975,11 @@ def async_update_device(  # noqa: C901
                 new_values["identifiers"] = old_identifiers | merge_identifiers
                 old_values["identifiers"] = old_identifiers
 
+        if not new_connections and not new_identifiers:
+            raise HomeAssistantError(
+                "A device must have at least one of identifiers or connections"
+            )
+
```

Comment:
This feels like it's a sanity check of the inputs.

Is there a reason it is not at the top of the function next to the others checks that raise similar Exceptions?

Reply:
No reason, you're right and I'll move it there 👍